sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -841,20 +853,150 @@ private void ensureToNotCalledYet() {
       return toBuilder().setFormatFunction(formatFunction).build();
     }
 
-    static class ConstantTableFunction<T> implements
-        SerializableFunction<ValueInSingleWindow<T>, TableDestination> {
+    /**
+     * Always returns a constant table destination.
+     */
+    static class ConstantTableDestinations<T> extends DynamicDestinations<T, TableDestination> {
```

Comment:
Was thinking that. Should I move into the generic BigQueryHelpers class, or into a new class?